infra/Pos/Diffusion/Subscription/Dns.hs

```diff
@@ -107,10 +113,21 @@ dnsSubscriptionWorker oq networkCfg DnsDomains{..} keepaliveTimer nextSlotDurati
 
     -- How long to wait before retrying in case no alternative can be
     -- subscribed to.
-    retryInterval :: m Millisecond
-    retryInterval = do
+    retryInterval :: SharedAtomicT m Int -> m (Maybe Millisecond)
+    retryInterval retryCounter = do
+        counter <- modifySharedAtomic retryCounter (\c -> return (c + 1, fromIntegral c))
```

Comment:
This counter contains the number of times we've retried. Do we ever reset it to 0?

Reply:
As we talked on slack, let's reset the counter the same way that the subscription timer get reset.  Is it fine to add a `retryCounter :: SharedAtomicT d Int` to the `Logic`.  It cannot be created in `runDiffusionLayerFull` because there we don't have access to the monad `d`.  Otherwise I can just use an `MVar` instead of a `SharedAtomicT`.

Reply:
I feel this implementation detail should not be exposed in the logic layer, and anyway we want to get rid of `Mockable` so I prefer to crate an `MVar` in `runDiffusionLayerFull`.